File: src/types/session.ts
export interface StudySession {
  id: string
  startedAt: string
  endedAt: string
  plannedDurationSeconds: number
  actualDurationSeconds: number
  completed: boolean
  mode: 'focus' | 'break' | 'deep'
  note?: string
}

export interface CreateSessionPayload {
  id: string
  startedAt: string
  endedAt: string
  plannedDurationSeconds: number
  actualDurationSeconds: number
  completed: boolean
  mode: 'focus' | 'break' | 'deep'
  note?: string
}

export interface StudyStats {
  todayMinutes: number
  todaySessionsCount: number
  weekMinutes: number
}

File: src-tauri/src/models/session.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize, sqlx::FromRow)]
pub struct StudySession {
    pub id: String,
    pub started_at: String,
    pub ended_at: String,
    pub planned_duration_seconds: i64,
    pub actual_duration_seconds: i64,
    pub completed: i64, // SQLite uses 0/1 for boolean
    pub mode: String,
    pub note: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct CreateSessionPayload {
    pub id: String,
    pub started_at: String,
    pub ended_at: String,
    pub planned_duration_seconds: i64,
    pub actual_duration_seconds: i64,
    pub completed: bool,
    pub mode: String,
    pub note: Option<String>,
}
